user_guide_src/source/outgoing/view_cells.rst

```diff
@@ -56,6 +64,176 @@ When you use it this way, all of the parameters must always be specified in the
         return view('recentPosts', ['posts' => $posts]);
     }
 
+.. _controlled-cells:
+
+****************
+Controlled Cells
+****************
+
+Controlled Cells have two primary goals: to make it as fast as possible to build the cell, and provide additional logic and flexibility to your views, if they need it. The class must extend ``CodeIgniter\View\Cells\Cell``. They should have a view file in the same folder. By convention the class name should be PascalCase and the view should be the snake_cased version of the class name. So, for example, if you have a ``MyCell`` class, the view file should be ``my_cell.php``.
```

Comment:
I hadn't noticed that since on non-case-sensitive file-systems they do display together. At least it does on my Mac, and I assume Windows would do the same thing. Looking at GitHub's file listings and it doesn't, though, which I assume is it running on Linux. 

I'm not sure the best solution here. By convention, views in CI have always been lowercase, which is why I went that route. They've also never needed special extensions other than `.php`. 

We could pull a book out of Next and SvelteKit and require them to be in their own folder, but that seems like unnecessary complexity. 

While it would be ideal if it was next to each other in all situations if I'm being honest, I'm not too concerned. In typical MVC setups they're not next to each other either, Controllers and Views that is. I realize it's not a perfect comparison, too....

Reply:
I was working on the generator command for Cells tonight and it seems the generator command will automatically force uppercase on the file name. Which gave me the happy accident of having the following files: 

```
- Test.php
- TestCell.php
```

While that goes against our current conventions with views being lower-cased, it's not a terrible thing to force the class to have the Cell postfix and the view be the name without. Though I'll be honest, it took me a moment to realize that `Test.php` was the view and not an artifact leftover from a previous run. :)

Reply:
I remember, the generator converts names like `name-name-tow` to `NameNametow`.

Reply:
> ... it seems the generator command will automatically force uppercase on the file name. 

When supplying the `name` argument to the generator command call, it is assumed that the `name` will be used for a class file. That's why the class name and file name are turned to PascalCase.

Reply:
From what I see, the only official package that uses view files (Shield) uses a `sneak_case` convention. So in my opinion, it would be a good idea to continue this.

Although I'm not going to hide the fact that I've also always used this convention for views.

@paulbalandan Are there any chances to easily implement the configuration of this from the command class level? Could we add something like `depascalize` for the class name only?

```
function depascalize(string $string): string
{
    return strtolower(substr(preg_replace('/([A-Z])/', '_$1', $string), 1));
}
```

Reply:
If I will be making a `make:cell` command, I would add a `getView()` method where I would "depascalize" the file path. My assumption is this `make:cell` command will produce two files, the custom `Cell` and its associated view file made thru `getView()`.